core/consensus/pow/src/lib.rs

```diff
@@ -157,6 +173,10 @@ impl<C, Algorithm> PowVerifier<C, Algorithm> {
 	{
 		const MAX_TIMESTAMP_DRIFT_SECS: u64 = 60;
 
+		if *block.header().number() < self.check_inherents_after {
```

Comment:
Why did we need to add this? Do we have to deal with any existing deployment?

Reply:
I added this parameter because runtime upgrades may break older inherent data check. For example, when the inherent data format is changed completely or an old inherent module is removed. In those cases, some chains might want to completely skip the check, because the older the block is, the more "finalized" it is because of PoW built on top of it.

This branch can be disabled by simply setting `check_inherents_after` to 0.